R/derive_advs_params.R

```diff
@@ -0,0 +1,317 @@
+#'   "01-101-1002", "HIPCIR", "Hip Circumference (cm)", 132, "cm", "WEEK 3",
+#'   "01-101-1002", "WSTCIR", "Waist Circumference (in)", 47.24, "in", "SCREENING",
+#'   "01-101-1002", "WSTCIR", "Waist Circumference (in)", 46.46, "in", "WEEK 2",
+#'   "01-101-1002", "WSTCIR", "Waist Circumference (in)", 46.06, "in", "WEEK 3"
+#' )
+#'
+#' derive_param_waisthip(
+#'   advs,
+#'   by_vars = exprs(USUBJID, VISIT),
+#'   wstcir_code = "WSTCIR",
+#'   hipcir_code = "HIPCIR",
+#'   set_values_to = exprs(
+#'     PARAMCD = "WAISTHIP",
+#'     PARAM = "Waist to Hip Ratio"
+#'   ),
+#'   get_unit_expr = admiral::extract_unit(PARAM)
+#' )
+derive_param_waisthip <- function(dataset,
+                                  by_vars,
+                                  wstcir_code = "WSTCIR",
+                                  hipcir_code = "HIPCIR",
+                                  set_values_to = exprs(PARAMCD = "WAISTHIP"),
+                                  filter = NULL,
+                                  get_unit_expr = NULL) {
+  assert_vars(by_vars)
+  assert_data_frame(dataset, required_vars = exprs(!!!by_vars, PARAMCD, AVAL))
+  assert_character_scalar(wstcir_code)
+  assert_character_scalar(hipcir_code)
+  assert_varval_list(set_values_to, required_elements = "PARAMCD")
+  assert_param_does_not_exist(dataset, set_values_to$PARAMCD)
+  filter <- assert_filter_cond(enexpr(filter), optional = TRUE)
+  get_unit_expr <- assert_expr(enexpr(get_unit_expr), optional = TRUE)
+
+  derive_param_ratio(
+    dataset,
+    filter = !!filter,
+    numerator_code = wstcir_code,
+    denominator_code = hipcir_code,
+    by_vars = by_vars,
+    set_values_to = set_values_to,
+    get_unit_expr = !!get_unit_expr
+  )
+}
+
+#' Adds a Parameter for Waist to Height Ratio
+#'
+#' @description Adds a record for Waist to Height Ratio using Waist Circumference and Height
+#' each by group (e.g., subject and visit) where the source parameters are available.
+#'
+#' **Note:** This is a wrapper function for the more generic
+#' \code{admiral::derive_param_computed()}.
+#'
+#' @param dataset Input dataset
+#'
+#'   The variables specified by the \code{by_vars} argument are expected to be in the dataset.
+#'   \code{PARAMCD}, and \code{AVAL} are expected as well.
+#'
+#'   The variable specified by \code{by_vars} and \code{PARAMCD} must be a unique key of
+#'   the input dataset after restricting it by the filter condition (\code{filter} argument)
+#'   and to the parameters specified by \code{wstcir_code} and \code{height_code}.
+#'
+#' @param wstcir_code Waist Circumference parameter code
+#'
+#'   The observations where \code{PARAMCD} equals the specified value are considered
+#'   as the Waist Circumference
+#'
+#'   *Permitted Values:* character value
+#'
+#' @param height_code Height parameter code
+#'
+#'   The observations where \code{PARAMCD} equals the specified value are considered
+#'   as the Height. It is expected that Height is measured in cm
```

Comment:
```suggestion
#'   as the Height. It is expected that Height is measured in cm.
```

Reply:
But can you have height in other units (m)? Can we add flexibility here?

Reply:
Thanks for your comment. This indeed became obsolete after unit conversion on the fly has been implemented. Will remove it in the following commit.

Reply:
Done